src/transformers/generation/stopping_criteria.py

```diff
@@ -372,9 +372,9 @@ def _stop_string_create_embedding_vec(token_list, token_indices, stop_strings) -
         token_valid_positions, token_end_overlaps = StopStringCriteria._stop_string_get_matching_positions(
             token_list, token_indices, stop_strings
         )
-
+        # Insert a dummy 1 to catch the case where there are no valid internal positions for any stop string
         max_valid_positions = max(
-            len(val) for positions in token_valid_positions.values() for val in positions.values()
+            [len(val) for positions in token_valid_positions.values() for val in positions.values()] + [1]
         )
```

Comment:
I have a slight preference for something like `max_valid_positions = 1 if len(token_valid_positions)==0 else max(...)`, but same same

Reply:
Done! And comment added to the `false_strings` in the test